app/controllers/application_controller.rb

```diff
@@ -6,4 +6,11 @@ def check_logged_user
 
     redirect_to login_path
   end
+
+  def find_current_cart
+    cart = Cart.where('user_id = ? AND placed_order = ?', current_user.id, false).first
+    return @current_cart ||= cart unless cart.nil?
+
+    @current_cart = Cart.create(user_id: current_user.id)
```

Comment:
Also, you can set the cart variable directly as `@current_cart` and you won't need the return line anymore, as you can write `@current_cart = current_user.carts.create if @current_cart.nil?`